src/lib/components/identity/identity-details.svelte

```diff
@@ -55,15 +66,17 @@
                 <div class="fs-4 fw-bold">{identity?.username}</div>
                 <div class="text-secondary fw-bolder mt-1 text-break">{identity?.id}</div>
             </div>
-            <div class="d-flex flex-row justify-content-end role-padding">
-                <div class="pb-5">
-                    {#if userRole === UserRoles.Admin}
-                    <Dropdown size="sm">
-                        <DropdownToggle {caret}>
+            <div class="pb-5 role-padding">
+                {#if userRole === UserRoles.Admin}
+                    <Dropdown size="sx">
+                        <DropdownToggle caret={!loading}>
                             {#if loading}
-                            <div class="ms-2 flex align-items-center">Updating role<Spinner size="sm" type="border" /></div>
-                            {:else}
-                            {identity?.role}
+                                <div class="ms-2 flex align-items-center">
+                                    Updating role<Spinner size="sm" type="border" />
+                                </div>
+                            {/if}
+                            {#if innerWidth >= 400}
```

Comment:
I think this can be done prettier with css media queries. Look [here](https://www.w3docs.com/snippets/css/how-to-hide-elements-in-a-responsive-layout.html#:~:text=To%20hide%20an%20element%20in,on%20devices%20smaller%20than%20767px.) how to hide an element on a specific view width.